extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetTableLocation.java

```diff
@@ -91,15 +92,15 @@ public ParquetTableLocation(@NotNull final TableKey tableKey,
 
         final int rowGroupCount = rowGroupIndices.length;
         rowGroups = IntStream.of(rowGroupIndices)
-                .mapToObj(rgi -> parquetFileReader.fileMetaData.getRow_groups().get(rgi))
-                .sorted(Comparator.comparingInt(RowGroup::getOrdinal))
-                .toArray(RowGroup[]::new);
-        final long maxRowCount = Arrays.stream(rowGroups).mapToLong(RowGroup::getNum_rows).max().orElse(0L);
+                .mapToObj(rgi -> parquetMetadata.getBlocks().get(rgi))
```

Comment:
This is also a change; previously the block metadata was coming from the file instead of the metadata file.

Reply:
I'm concerned about this. I think this is a merged list of blocks, so we might be including all the row groups from all the files, once per file.